core/arch/arm/include/mm/core_mmu.h

```diff
@@ -345,6 +347,36 @@ static inline size_t core_mmu_get_block_offset(
 }
 
 /*
+ * core_mmu_is_vaspace() - Check if memory region belongs to
+ *  empty virtual address space that is used for dymanic mappings
+ * @mm:		memory region to be checked
+ * @returns result of the check
+ */
+static inline bool core_mmu_is_vaspace(struct tee_mmap_region *mm)
+{
+	return mm->type == MEM_AREA_RES_VASPACE ||
+		mm->type == MEM_AREA_SHM_VASPACE;
+}
+
+/*
```

Comment:
Yep, now it should be in the right place.
Regarding name: I think that `core_mmu_is_dynamic_vaspace()` is most appropriate. But it is somewhat long. Anyways, I'll rename it.